modules/ding_gatewayf/modules/ding_gatewayf_registration/ding_gatewayf_registration.module

```diff
@@ -0,0 +1,456 @@
+  // the DDB default theme. The module existing_pages could have been used to
+  // wrapper the form in panels. But seen form a performance point this is
+  // faster and simpler.
+  $form = array(
+    '#prefix' => '<div class="empty-sidebars default-layout"><div class="layout-wrapper"><div class="pane-content"><div class="primary-content">',
+    '#suffix' => '</div></div></div></div>',
+  );
+
+  $form['accept'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('I accept the user agreements'),
+    '#description' => $config['acceptance']['description'],
+    '#required' => TRUE,
+  );
+
+  $form['name'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Name'),
+    '#description' => t('You full name.'),
+    '#maxlength' => 100,
+    '#required' => TRUE,
+    '#attributes' => array(
+      'autocomplete' => 'off',
+    ),
+  );
+
+  $form['mail'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Email address'),
+    '#description' => t('The Email address that we can use to get in contact with you.'),
+    '#required' => TRUE,
+    '#attributes' => array(
+      // Auto-complete is disabled here as the registration may be preform on a
+      // public computer in the library and information about which user there
+      // is registered and the mail address is classified information.
+      'autocomplete' => 'off',
+    ),
+  );
+
+  // Get branch options form the active provider.
+  global $user;
+  $provider = _ding_provider_get_provider('reservation');
+  $options = ding_provider_invoke('reservation', 'options', $user);
+  $form['branch'] = $options[$provider['module'] . '_preferred_branch'];
+
+  $form['pin'] = array(
+    '#type' => 'password',
+    '#title' => t('Pin code'),
+    '#description' => t('Pin code with an max length of %length', array('%length' => ding_user_get_pincode_length())),
+    '#size' => 10,
+    '#maxlength' => ding_user_get_pincode_length(),
+    '#required' => TRUE,
+    '#attributes' => array(
+      'autocomplete' => 'off',
+    ),
+  );
+
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Create me'),
+  );
+
+  // Link cancel button to gatewayf logout to ensure that the user is signed
+  // out correctly.
+  $form['cancel'] = array(
+    '#markup' => l(t('Cancel'), DING_GATEWAYF_LOGOUT_URL, array('attributes' => array('class' => 'ding-gatewayf--cancel'))),
+  );
+
+  $form['#submit'] = array('ding_gatewayf_registration_acceptance_submit');
+
+  return $form;
+}
+
+/**
+ * Submit handler for the acceptance form.
+ *
+ * It tires to create the user at the provider and redirect the user.
+ */
+function ding_gatewayf_registration_acceptance_submit($form, $form_state) {
+  $values = $form_state['values'];
+
+  // This value was not store in the form to ensure that the user can reload the
+  // form.
+  $required_attributes = $_SESSION[DING_GATEWAYF_ATTRIBUTES];
+  unset($_SESSION[DING_GATEWAYF_ATTRIBUTES]);
```

Comment:
If we cannot store it in the form should we at least add it to an access check when entering this flow?